common/djangoapps/course_category/models.py

```diff
@@ -0,0 +1,74 @@
+from django.db import models
+from django.db.models.signals import m2m_changed, pre_delete, post_delete
+from django.dispatch import Signal, receiver
+from django.utils.translation import ugettext_lazy as _
+
+from mptt.models import MPTTModel, TreeForeignKey
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
+
+from .tasks import task_reindex_courses
+
+
+move_to = Signal()
+
+
+class CourseCategory(MPTTModel):
+    name = models.CharField(max_length=255, verbose_name=_("Category Name"), unique=True)
+    description = models.TextField(null=True, blank=True)
+    img = models.ImageField(upload_to='course_category', blank=True)
+    parent = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
+    enabled = models.BooleanField(default=True)
+    slug = models.SlugField(max_length=255, unique=True)
+    url = models.URLField(max_length=200, null=True, blank=True)
+    courses = models.ManyToManyField("course_overviews.CourseOverview")
+
+    class MPTTMeta:
+        order_insertion_by = ['name']
+
+    class Meta:
+        verbose_name_plural = 'Course Categories'
+
+    def __unicode__(self):
+        return self.name
+
+    def move_to(self, target, position='first-child'):
+        self._tree_manager.move_node(self, target, position)
+        move_to.send(sender=self.__class__, instance=self)
+
+    @classmethod
```

Comment:
Is it really a `classmethod`?

Will continue review from this point tomorrow.

Reply:
I even have no idea the usage case of the method 🙂It even does not use.
`classmethod` why not? As one of the potential solution approach. Let it be 🙂